docs/documentation/automation/question_exec.md

```diff
@@ -6,4 +6,105 @@ sidebar_label: Question Exec
 import useBaseUrl from '@docusaurus/useBaseUrl'; 
 import Highlight from '@theme/Highlight';
 
-# Subtitle {#subtitle}
+<span class="hero__subtitle">
+
+_Customize Survey Questions with Javascript Code_
+
+</span>
+
+## Overview
+Surveys are among Cotalker's main features. Their flexibility and customization ability allow them to adapt to each business's needs.
+
+This spec expands the notion of survey functionality by introducing _Question EXEC_, custom code execution for your survey questions.
+
+:::info
+- Custom code must be written in **Javascript**.
+- Code should always be previously _sandboxed_.
+:::
+
+## Set Up
+
+To add your custom code, choose any survey component and select the **Automation** tab. The code is added in the field at the bottom of the settings panel.
+_See the image below:_
+
+<img alt="automation tab" class="img_sizing_narrow item shadow--md" src={useBaseUrl('img/automation_question_exec_00.png')} />
+<br/>
+<br/>
+
+You can add custom code in any of the four _stages_ of a _survey component's lifecycle_. Set the **stage** either with the dropdown menu or by clicking the corresponding icon. 
+_See the image below:_
+
+<img alt="automation tab" class="img_sizing_narrow item shadow--md" src={useBaseUrl('img/automation_question_exec_01.png')} />
+<br/>
+
+## Component Lifecycle Stages
+- **Preload**: The code executes when a new survey instance is summoned.
```

Comment:
This might me tricky, because it's only client side on creaton?

Reply:
This image says "client and server". 

Does "at Creation" mean that "preload" is when the survey is created and saved in the Admin section?

![Screen Shot 2021-04-06 at 20 37 12](https://user-images.githubusercontent.com/47286214/113794391-d1d3d300-9720-11eb-9026-b5fa1cfa1cb2.png)

Reply:
The wiki says:

> # preload: Array<COTCommand<null>>
> Preload must run when:
> - A new survey is built in a channel (clients)
> - A new task is been created (clients)
> - A new is instanced before sent (server)

The last two are unclear to me.

Reply:
I think only the first one is implemented.
@guillevalin can you double check ?